numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -1884,8 +1884,32 @@ gentype_reduce(PyObject *self, PyObject *NPY_UNUSED(args))
     PyTuple_SET_ITEM(ret, 0, obj);
     obj = PyObject_GetAttrString((PyObject *)self, "dtype");
     if (PyArray_IsScalar(self, Object)) {
-        mod = ((PyObjectScalarObject *)self)->obval;
-        PyTuple_SET_ITEM(ret, 1, Py_BuildValue("NO", obj, mod));
+        PyObject* val = ((PyObjectScalarObject *)self)->obval;
+        PyObject * tup = Py_BuildValue("NO", obj, val);
+        if (tup == NULL) {
+            return NULL;
+        }
+        PyTuple_SET_ITEM(ret, 1, tup);
+    }
+    else if (obj && PyDataType_FLAGCHK((PyArray_Descr *)obj, NPY_LIST_PICKLE)) {
+        /* a structured dtype with an object in a field */
+        PyObject *arr = PyArray_FromScalar(self, NULL);
+        if (arr == NULL) {
+            return NULL;
+        }
+        PyArrayIterObject *it = (PyArrayIterObject *)PyArray_IterNew(arr);
+        if (it == NULL) {
+            return NULL;
+        }
+        /* get arr[0] */
```

Comment:
```suggestion
        /* get arr.item() */
```
or perhaps
```suggestion
        /* get arr.item[()] */
```

Reply:
^ corrected

Reply:
adopting the latter

Reply:
Thought about it some more - this is actually the former, since the latter is just `self`

Reply:
so arr.item() ?